File: api-gateway-core/src/main/java/top/codelong/apigatewaycore/common/HttpStatement.java
package top.codelong.apigatewaycore.common;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import top.codelong.apigatewaycore.enums.HTTPTypeEnum;

/**
 * http请求信息
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class HttpStatement {
    private String interfaceName;
    private String methodName;
    private String[] parameterType;
    private Boolean isAuth;
    private Boolean isHttp;
    private HTTPTypeEnum httpType;
    private String serviceId;

    public String getServiceId() {
        // 如果 serviceId 为空，使用接口名作为服务ID
        return serviceId != null ? serviceId : interfaceName;
    }
}

File: api-gateway-core-go/internal/model/entity/http_statement.go
package entity

import "api-gateway-core-go/internal/model/enum"

type HttpStatement struct {
	InterfaceName string        `json:"interfaceName"`
	MethodName    string        `json:"methodName"`
	ParameterType []string      `json:"parameterType"`
	IsAuth        bool          `json:"isAuth"`
	IsHttp        bool          `json:"isHttp"`
	HttpType      enum.HTTPType `json:"httpType"`
	ServiceId     string        `json:"serviceId"`
}

func (h *HttpStatement) GetServiceId() string {
	if h.ServiceId != "" {
		return h.ServiceId
	}
	return h.InterfaceName
}
